jupyter_server/services/contents/handlers.py

```diff
@@ -136,26 +136,27 @@ async def get(self, path=""):
             raise web.HTTPError(400, "Content %r is invalid" % content_str)
         content = int(content_str or "")
 
-        md5_str = self.get_query_argument("md5", default="0")
-        if md5_str not in {"0", "1"}:
-            raise web.HTTPError(400, "Content %r is invalid" % md5_str)
-        md5 = int(md5_str or "")
+        sha256_str = self.get_query_argument("sha256", default="0")
+        if sha256_str not in {"0", "1"}:
+            raise web.HTTPError(400, "Content %r is invalid" % sha256_str)
+        sha256 = int(sha256_str or "")
 
         if not cm.allow_hidden and await ensure_async(cm.is_hidden(path)):
             await self._finish_error(
                 HTTPStatus.NOT_FOUND, f"file or directory {path!r} does not exist"
             )
+        kwargs = {
+            "path": path,
+            "type": type,
+            "format": format,
+            "content": content,
+        }
+        if cm.support_sha256:
+            kwargs["sha256"] = sha256
```

Comment:
If the `ContentsManage` does not declare itself to support sha256(by setting `support_sha256` to `True`), handler will never pass `sha256` argument to the `ContentsManager.get`